src/main/kotlin/com/github/jyoo980/reachhover/ui/ReachabilityButton.kt

```diff
@@ -0,0 +1,60 @@
+package com.github.jyoo980.reachhover.ui
+
+import com.github.jyoo980.reachhover.MyBundle
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiIdentifier
+import java.awt.Desktop
+import java.net.URI
+import javax.swing.JButton
+import javax.swing.SwingConstants
+
+sealed class ReachabilityButton(element: PsiElement) {
+
+    private val elementUnderCursor = element
+    // TODO: Either come up with a custom icon, or find a way to resize (QuestionDialog is 32x32, we
+    // need 16x16).
+    val ui: JButton =
+        JButton(AllIcons.General.QuestionDialog).also {
+            it.horizontalAlignment = SwingConstants.LEFT
+            it.isBorderPainted = false
+            it.isContentAreaFilled = false
+        }
+
+    abstract fun setButtonText(optText: String? = null)
+
+    open fun activateAction(editor: Editor) {
+        // TODO: stub behaviour, link this to actual reachability action later. Should be an
+        // abstract fun once we're done.
+        this.ui.addActionListener {
+            Desktop.getDesktop().browse(URI("https://www.youtube.com/watch?v=dQw4w9WgXcQ"))
+        }
+    }
+
+    protected fun optIdentifierName(): String? {
```

Comment:
I don't think this `opt`-prefix thing is necessary since we can always check the type of the variable, and see the question mark. There will be a lot of optional values, and these `opt`-things might become an `opt`-hell at some point.